onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -1260,15 +1262,25 @@ class GemmOpSupportChecker : public BaseOpSupportChecker {
       const InitializedTensorSet& /* initializers */, const NodeUnit& node_unit,
       const OpSupportCheckParams& /* params */) const override;
   int GetMinSupportedOpSet(const NodeUnit& node_unit) const override;
+
+  bool IsNodeUnitTypeSupported(const NodeUnit& /* node_unit */) const override { return true; }
+
+  bool IsQuantizedOp(const NodeUnit& node_unit) const override;
 };
 
+bool GemmOpSupportChecker::IsQuantizedOp(const NodeUnit& node_unit) const {
+  const auto quant_type = GetQuantizedOpType(node_unit);
+  return quant_type == QuantizedOpType::QDQGemm || quant_type == QuantizedOpType::QLinearMatMul;
+}
+
 bool GemmOpSupportChecker::HasSupportedInputOutputsImpl(
     const InitializedTensorSet& initializers, const NodeUnit& node_unit,
     const OpSupportCheckParams& params) const {
-  if (node_unit.OpType() != "QLinearMatMul")
+  if (!IsQuantizedOp(node_unit)) {
     return BaseOpSupportChecker::HasSupportedInputOutputsImpl(initializers, node_unit, params);
+  }
 
-  // QLinearMatMul
+  // QLinearMatMul/QDQGemm
   if (!HasValidBinaryOpQuantizedInputTypes(node_unit))
```

Comment:
You will also need to check if the bias is an initializer for qdqgemm, like this
https://github.com/microsoft/onnxruntime/blob/19464614e7a71c42016dcf0c5bc3b9e7880975db/onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc#L1127-L1131

Reply:
Why does the bias need to be an initializer? We check the shape to make sure it's 1D but as it doesn't need to be transposed can't we accept a non-initializer input at runtime?

Reply:
not sure - but I guess maybe we also need special handling of bias of QDQ Gemm to add scales for quantization input  just as the case for qlinearconv/qdqconv like these lines here so we need it to be an initializer?
https://github.com/microsoft/onnxruntime/blob/c5de493a8a6a62d07f7ec32cebe784765826ede4/onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc#L1342-L1356

Reply:
ok - I think I understand it now. the nnapi spec says 

> For input tensor of ANEURALNETWORKS_TENSOR_QUANT8_ASYMM and ANEURALNETWORKS_TENSOR_QUANT8_ASYMM_SIGNED, the bias should be of ANEURALNETWORKS_TENSOR_INT32, with zeroPoint of 0 and bias_scale == input_scale * filter_scale.

which I believe is similar for Conv and why we do some manual processing of the bias input to apply that scaling. It's somewhat confusing that it's called `filter_scale` as it's the weights input (it's the filter input in Conv so makes more sense there), but based on the tflite impl they use the name `filter` for the weights input.
 https://chromium.googlesource.com/external/github.com/tensorflow/tensorflow/+/refs/tags/v2.8.0/tensorflow/lite/kernels/fully_connected.cc#203

Reply:
I see. thx for the explanation. 

searched a bit online... some ppl says the weight matrix of a fully connected layer can actually be reshaped into a conv layer filter, probably the naming is not so irrelevant. don't know:)